mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/mailbox/vault/SupportedChecks.java

```diff
@@ -0,0 +1,158 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.vault;
+
+import static org.apache.james.mailbox.vault.GenericComponents.OperatorSign.CONTAINS;
+import static org.apache.james.mailbox.vault.GenericComponents.OperatorSign.GREATER_OR_EQUAL;
+import static org.apache.james.mailbox.vault.GenericComponents.OperatorSign.LESS_THAN_OR_EQUAL;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.apache.james.mailbox.vault.GenericComponents.FieldName;
+import org.apache.james.mailbox.vault.GenericComponents.OperatorSign;
+import org.apache.james.mailbox.vault.GenericComponents.Value;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableTable;
+
+public class SupportedChecks {
+
+    static class ComparedValue extends Value {
```

Comment:
it will be generated from the layer near to the DAO

Reply:
What's the added value of this compared to `s/value/Field` ?